patterns/1-initial/transitioning-contractor-code-to-innersource-model.md

```diff
@@ -0,0 +1,84 @@
+## Problem
+
+Contractor developers are often not motivated (through forces described below) to not engage in InnerSource activities. Once delivered, and even if the code is made visible, their projects are often less likely to be part of successful InnerSourced engagements.
+
+## Context
+
+This problem exists where an organization either:
+
+- out-sources the development of a well defined project or
+- engages external firms for staff augmentation and has mixed teams of permanent employees with a large percentage of contract staff. This issue may occur less often in a mixed team where InnerSource principles and ways of working are already established within the organization, but the forces described below are still often at play.
+
+## Forces
+
+Contractor Motivation and Constraints:
+
+- Contracts with third-party developers tend to be focused on delivering an end result in the fastest possible fashion. As a result, all InnerSource activities (e.g. responding to third-party PRs) are considered to be distractions or something that will “slow down” delivery.
+- Concern that accepting code from other parts of the business might introduce security risks, scope creep or other issues that would subsequently have to be resolved by the contract team.
+- Concern that accepting PRs from other parts of the company may “muddy the waters” when it comes to assessing what parts of the project were completed/delivered by the contracted developers.
+- Contractors may have concerns about loss of control over the projects technical details if other teams are contributing significantly.
+Even when motivations align (a project can be completed faster or with higher quality by engaging in InnerSource) contractors might still be unclear if they can because it is not explicitly listed as allowed in the contract terms.
+
+All of the above can mean that even if an individual contract developer wants to engage in InnerSource, there are system-level constraints pushing them not to.
+
+It should be noted that the above scenario is indirectly impacted by:
+
+- The norms around defining Statements of Work for third party contractors
+- Pressures to reduce contractor costs during procurement
+- Ability to tie contributions to payment at a granular level.
+
+It could also be noted that the contractor motivations in this instance is almost like a more extreme instance of the oft-reported organizational/budgetary constraints that might exist for some internal business units. (Not sure if this is relevant, but it does seem to be an extreme case of what is reported as common objections even in internal teams).
+
+## Solutions
+
+At the outset of the project, clearly define:
+
+- That the code will be made visible by default.
+- That the architecture of the code should be modular and ready for component reuse.
+- How the project will be transitioned to an InnerSource project. This could be similar to a transition of ownership plan for an open source project which should include: 
```

Comment:
```suggestion
- How the project will be transitioned to an InnerSource project. This could be similar to a transition of ownership plan for an open source project which should include:
```